sdk/src/types/api/core/response.rs

```diff
@@ -72,11 +85,95 @@ pub struct StatusResponse {
     serde(rename_all = "camelCase")
 )]
 pub struct MetricsResponse {
+    #[serde(with = "crate::utils::serde::string")]
     pub blocks_per_second: f64,
+    #[serde(with = "crate::utils::serde::string")]
     pub confirmed_blocks_per_second: f64,
+    #[serde(with = "crate::utils::serde::string")]
     pub confirmed_rate: f64,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ProtocolParametersResponse {
+    parameters: Box<[ProtocolParameters]>,
+    version_map: BTreeMap<u8, usize>,
+    epoch_map: BTreeMap<EpochIndex, usize>,
+}
+
+impl ProtocolParametersResponse {
+    pub fn iter(&self) -> impl Iterator<Item = (EpochIndex, &ProtocolParameters)> {
+        self.epoch_map.iter().map(|(&epoch, &i)| (epoch, &self.parameters[i]))
+    }
+
+    pub fn latest(&self) -> &ProtocolParameters {
```

Comment:
That wouldn't work if the array contain future parameters?

Reply:
I'm under the impression they would be ordered by start epoch so we would have to iterate until we find one where current epoch is between two paremeters

Reply:
But we don't know the current epoch unless we have the parameters